components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.ts

```diff
@@ -33,6 +41,23 @@ export class CreateObjectModalComponent implements OnInit {
     });
   }
 
+  ngOnChanges(changes: SimpleChanges): void {
+    // if a new list of projects to populate dropdown with is passed in we update the dropdown
+    const checked = false;
+    if (hasIn('assignableProjects.currentValue', changes)) {
+      changes.assignableProjects.currentValue.forEach((proj: Project) =>
+        this.projects[proj.id] = { ...proj, checked });
+    }
+  }
+
+  onProjectChecked(project: ProjectChecked): void {
+    this.projects[project.id] = project;
+  }
```

Comment:
Hmm... wondering if you identified some edge case where you need to replace the whole project? Seems better to do what only what you are claiming:
```suggestion
    this.projects[project.id].checked = project.checked;
```